packages/api-client/src/tcp/WebSocketClient.ts

```diff
@@ -111,12 +110,10 @@ export class WebSocketClient extends EventEmitter {
       // before we try any connection, we first refresh the access token to make sure we will avoid concurrent accessToken refreshes
       await this.refreshAccessToken();
     }
-    // Note: Do NOT await `onConnect` otherwise the websocket will not connect during notification stream processing
-    void this.onConnect();
```

Comment:
The problem lies here in the fact that we would call the `onConnect` **before** actually being connected to the websocket. 
Which means that we start processing the notification stream but there could be messages that come in between the moment we start processing the notification stream and the moment the websocket is actually connected (and thus we have potential missing messages)